src/composables/Project.js

```diff
@@ -237,3 +237,24 @@ export async function checkout(projectId, branch) {
   });
   return GitEvent.CheckoutEvent.next();
 }
+
+/**
+ * Create branch from another
+ * @param {String} projectId - Id of project.
+ * @param {String} newBranch - New branch name.
+ * @param {String} branch - Branch name.
+ * @param {Boolean} haveToCheckout - Indicate if we have to checkout on new branch.
+ * @return {Promise<void>} Promise with nothing on success otherwise an error.
+ */
+export async function newBranchFrom(projectId, newBranch, branch, haveToCheckout) {
+  await git.branch({
+    fs,
+    dir: `/${projectId}`,
+    ref: newBranch,
+    object: branch,
+  });
+  if (!haveToCheckout) {
+    return GitEvent.CheckoutEvent.next();
```

Comment:
if you don't have to checkout, then send a checkout event... why ? what for ?

Reply:
Because, to create a branch on git we use `git checkout -b name`, so it's also a checkout.